unity-renderer/Assets/Scripts/MainScripts/DCL/UserProfile/UserProfile.cs

```diff
@@ -39,27 +41,76 @@ public enum EmoteSource
     public AvatarModel avatar => model.avatar;
     public int tutorialStep => model.tutorialStep;
 
-    internal Dictionary<string, int> inventory = new Dictionary<string, int>();
+    internal Dictionary<string, int> inventory = new ();
 
     public ILazyTextureObserver snapshotObserver = new LazyTextureObserver();
     public ILazyTextureObserver bodySnapshotObserver = new LazyTextureObserver();
 
-    internal UserProfileModel model = new UserProfileModel() //Empty initialization to avoid nullchecks
+    internal static UserProfile ownUserProfile;
+
+    // Empty initialization to avoid null-checks
+    internal readonly UserProfileModel model = new () { avatar = new AvatarModel() };
+
+    private UserProfileModel ModelFallback()
+    {
+        var fallbackId = $"{FALLBACK_NAME}_{this.GetInstanceID()}";
+
+        UserProfileModel fallback = new UserProfileModel
+        {
+            // Required fields (otherwise exceptions will be thrown by OnUpdate subscribers)
+            userId = fallbackId,
+            name = FALLBACK_NAME,
+            description = "There was a problem with loading this profile. This is a fallback profile",
+
+            avatar = AvatarFallback(),
+
+            // Not-required/exceptions-free fields
+            ethAddress = fallbackId,
+            email = fallbackId,
+            baseUrl = fallbackId,
+        };
+
+        return fallback;
+    }
+
+    private AvatarModel AvatarFallback()
     {
-        avatar = new AvatarModel()
-    };
+        var fallbackId = $"{FALLBACK_NAME}_{this.GetInstanceID()}";
+
+        return new AvatarModel
+        {
+            id = fallbackId,
+            name = FALLBACK_NAME,
+            bodyShape = "urn:decentraland:off-chain:base-avatars:BaseMale",
+
+            skinColor = new Color(0.800f, 0.608f, 0.467f),
+            hairColor = new Color(0.596f, 0.373f, 0.216f),
+            eyeColor = new Color(0.373f, 0.224f, 0.196f),
+        };
+    }
 
     private int emoteLamportTimestamp = 1;
 
     public void UpdateData(UserProfileModel newModel)
     {
         if (newModel == null)
         {
-            model = new UserProfileModel();
-            return;
+            if (DataStore.i.featureFlags.flags.Get().IsFeatureEnabled("user_profile_null_model_exception"))
+                Debug.LogError("Model is null when updating UserProfile! Using fallback or previous model instead.");
```

Comment:
Just to confirm, this sends a Sentry error?

Reply:
I confirmed it with Ashley and it should work (regarding [this Sentry documentation page for Unity](https://docs.sentry.io/platforms/unity/)). 
```

public class TestMonoBehaviour : MonoBehaviour
{
    private GameObject testObject = null;

    void Start()
    {
        Debug.Log("Captured Log");              // Breadcrumb
        Debug.LogWarning("Captured Warning");   // Breadcrumb
        Debug.LogError("Captured Error");       // Captured Error

        // This will throw an unhandled Null Reference Exception
        testObject.GetComponent<Transform>();   // Captured error
    }
}
```